pkg/nutanix/client.go

```diff
@@ -254,7 +254,7 @@ func CheckResponse(r *http.Response) error {
 	}
 
 	pretty, _ := json.MarshalIndent(errRes, "", "  ")
-	return fmt.Errorf("error: %s", string(pretty))
+	return fmt.Errorf("status: %s, error: %s", r.Status, string(pretty))
```

Comment:
Another question here is whether we really want "pretty" i.e. indented json.

Reply:
With the indented format in the log, it's easier to read. As in the below example:

controller.go:303] demo-ocp-cluster-g1-pwbwx-worker-6vmhv: failed to check if machine exists: status: 401 UNAUTHORIZED, error-response: {
  "api_version": "3.1",
  "code": 401,
  "message_list": [
    {
      "details": "Basic realm=\"Intent Gateway Login Required\"",
      "message": "Authentication required.",
      "reason": "AUTHENTICATION_REQUIRED"
    }
  ],
  "state": "ERROR"
}